pkg/config/analysis/analyzers/analyzers_test.go

```diff
@@ -1019,6 +1019,28 @@ var testGrid = []testCase{
 			{msg.DestinationRuleSubsetNotSelectPods, "DestinationRule default/subsets-not-select-pods"},
 		},
 	},
+	{
+		name:       "DestinationRuleSubsetsWithTopologyLabels",
+		inputFiles: []string{"testdata/destinationrule-subsets-with-topology-labels.yaml"},
+		analyzer:   &destinationrule.PodNotSelectedAnalyzer{},
+		expected:   []message{
+			// Should not report false positives for topology labels
+			// All subsets should match because the analyzer augments pod labels with node topology labels:
+			// - "region-us-west": matches topology.kubernetes.io/region from node
+			// - "zone-us-west-1a": matches topology.kubernetes.io/zone from node
+			// - "app-v1": matches version label from pod
+			// - "mixed-labels": matches both version from pod AND topology.kubernetes.io/region from node
+		},
+	},
+	{
+		name:       "DestinationRuleEmptyTopologyLabels",
+		inputFiles: []string{"testdata/destinationrule-empty-topology-labels.yaml"},
+		analyzer:   &destinationrule.PodNotSelectedAnalyzer{},
+		expected: []message{
+			// Should report false positive for empty string topology label since AugmentLabels won't add empty values
```

Comment:
```suggestion
			// Istio doesn't match on empty node locality labels.
```
Regardless, there is no matching because the topology labels on the node aren't empty